organization/src/main/scala/com/improving/app/organization/repository/OrganizationByMemberProjectionHandler.scala

```diff
@@ -0,0 +1,303 @@
+package com.improving.app.organization.repository
+
+import akka.Done
+import akka.actor.typed.ActorSystem
+import akka.projection.eventsourced.EventEnvelope
+import akka.projection.scaladsl.Handler
+import com.improving.app.organization._
+import com.improving.app.organization.OrganizationEvent.Empty
+import com.improving.app.organization.domain.Organization._
+import org.slf4j.LoggerFactory
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class OrganizationByMemberProjectionHandler(tag: String, system: ActorSystem[_], repo: OrganizationRepository)
+    extends Handler[EventEnvelope[OrganizationEvent]]() {
+
+  private val log = LoggerFactory.getLogger(getClass)
+
+  implicit private val ec: ExecutionContext =
+    system.executionContext
+
+  override def process(envelope: EventEnvelope[OrganizationEvent]): Future[Done] = {
+    envelope.event match {
+      case Empty => Future.successful(Done)
+      case OrganizationEstablished(Some(orgId), info, parent, members, owners, contacts, actingMember, _) => {
+        log.info("OrganizationByMemberProjectionHandler: OrganizationEstablished")
+        val organization = Organization(
+          Some(orgId),
+          info,
+          parent,
+          members,
+          owners,
+          contacts,
+          Some(createMetaInfo(actingMember)),
+          OrganizationStatus.ORGANIZATION_STATUS_DRAFT
+        )
+        Future
+          .sequence(members.map(memberId => {
+            repo.updateOrganizationByMember(orgId.id, memberId.id, organization)
+          }))
+          .map(_ => Done)
+      }
+      case MembersAddedToOrganization(Some(orgId), newMembers, actingMember, _) => {
+        repo
+          .getOrganizationsByMemberByOrgId(orgId.id)
+          .map(organizations => {
+            Future.sequence(for {
+              organization <- organizations
+              member <- newMembers
+            } yield {
+              repo.updateOrganizationByMember(
+                organization.orgId.map(_.id).getOrElse("OrgId is NOT FOUND."),
+                member.id,
+                organization.copy(
+                  members = organization.members ++ newMembers,
```

Comment:
Most of the code here can be refactored into a single function - feel free to change the name of `handleEvent` to a better one if desired:

```
def handleEvent(newOrg: Organization) = repo
          .getOrganizationsByMemberByOrgId(orgId.id)
          .map(organizations => {
            Future.sequence(for {
              organization <- organizations
              member <- newMembers
            } yield {
              repo.updateOrganizationByMember(
                organization.orgId.map(_.id).getOrElse("OrgId is NOT FOUND."),
                member.id,
                newOrg
              )
            })
          })
          .map(_ => Done)

case MembersAddedToOrganization(Some(orgId), newMembers, actingMember, _) => {
  handleEvent(organization.copy(
    members = organization.members ++ newMembers,
    meta = Some(updateMetaInfo(organization.getMeta, actingMember))
  ))
}
```